src/ser/mod.rs

```diff
@@ -1329,6 +1342,29 @@ impl<'a, W: fmt::Write> ser::SerializeStruct for Compound<'a, W> {
 
         if !self.ser.compact_structs() {
             self.ser.indent()?;
+
+            if let Some((ref config, _)) = self.ser.pretty {
+                if let Some(ref field) = config.meta.field {
+                    // TODO: `self.ser.indent()` borrows the entire serializer mutably,
+                    // consider constraining the signature in the future to avoid this heap allocation.
+                    let doc_lines: Vec<_> = field
+                        .doc()
+                        .lines()
+                        .map(|line| {
+                            let mut buf = String::with_capacity(line.len() + 5);
+                            buf.push_str("/// ");
+                            buf.push_str(line);
+                            buf.push('\n');
+                            buf
+                        })
+                        .collect();
+
+                    for doc_line in doc_lines {
+                        self.ser.output.write_str(&doc_line)?;
+                        self.ser.indent()?;
```

Comment:
Note that the `indent` method overcaptures and borrows the entire serializer mutably (it only needs to borrow the output mutably), resulting in an avoidable heap allocation when we write the metadata. Resolving this would probably involve either inlining `indent` or making it a freestanding function, but that's outside of the scope of this PR, so I left it alone.